lib/fibonacci.rb

```diff
@@ -1,9 +1,27 @@
-# Improved Fibonacci
+# Improved Fibonacci????
+# PLEASE READ NOTE BELOW:
+# I didn't change anything from the class solution..not sure what to change!
 
 # Time Complexity - ?
 # Space Complexity - ?  (should be O(n))
 # Hint, you may want a recursive helper method
 # Hint:  Fib(0) = 0, Fib(1) = 1, work up from there
 def fibonacci(n)
```

Comment:
Lik you noted this is the class solution, but to find a specific fibonacci number, you only need to know the two previous.  Could you do this with 3 variables, instead of maintaining an array of all the numbers from 0 to n?